src/array.jl

```diff
@@ -383,6 +383,17 @@ end
     @inbounds A.refs[I...] = get!(A.pool, v)
 end
 
+function Base.fill(v::CategoricalValue{T, R}, dims::NTuple{N, Integer}) where {T, R, N}
+    A = CategoricalArray{T, N, R}(undef, dims)
+    merge_pools!(A, v, updaterefs=false)
+    fill!(A.refs, get!(A.pool, v))
```

Comment:
I understand that here although we use `get!` we are guaranteed that `v` is already in `A.pool` because we merged pools already - right?

Reply:
Yes. Anyway in practice `get!` will merge the levels if needed. The `merge_pools!` call is only there to ensure we don't recompute reference codes.

But actually your question made me realize there's a much simpler definition for this function (the current one is inspired by `fill!`, which is more complex). I've pushed a new commit.